subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/loader/DefaultToolingImplementationLoader.java

```diff
@@ -97,8 +98,11 @@ public ConsumerConnection create(Distribution distribution, ProgressLoggerFactor
             } else if (connection instanceof BuildActionRunner) {
                 adaptedConnection = new BuildActionRunnerBackedConsumerConnection(connection, modelMapping, adapter);
             } else {
-                return new UnsupportedOlderVersionConnection(connection, adapter);
+                throw unsupportedProvider();
             }
+
+            checkProviderVersion(adaptedConnection.getVersionDetails().getVersion());
```

Comment:
❌ The check shouldn't be eager at this point, but instead should only be done when the connection is actually used. You can wrap the connection in a deprecation logging adapter. That way we have access to the output streams that the user attached and can log the deprecation on those output streams. That way they are much more likely to actually see a warning.